plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -292,4 +333,70 @@ private static Block getColumnMetricBlock(RowType columnMetricType, Object min,
         rowBlockBuilder.closeEntry();
         return columnMetricType.getObject(rowBlockBuilder, 0);
     }
+
+    private static class StructLikeWrapperWithStructType
+    {
+        private final StructLikeWrapper structLikeWrapper;
+        private final Map<Integer, Integer> fieldIdToIndex;
+
+        public StructLikeWrapperWithStructType(StructLikeWrapper structLikeWrapper, Types.StructType structType)
+        {
+            this.structLikeWrapper = structLikeWrapper;
+            Map<Integer, Integer> fieldIdToIndex = new HashMap<>();
+            List<NestedField> fields = structType.fields();
```

Comment:
```
this.fieldIdToIndex = fields.stream().collect(Collectors.toMap(NestedField::fieldId, Function.identity()));
```

Reply:
This is incorrect. I need mapping from `fieldId -> its index`, your code gives `fieldId -> NestedField`, I could probably do it with some zipWithIndex method but I think standard way is more readible

Reply:
Sorry, i didn't pay enough attention here. Thanks for the explanation.